memfs/memory.go

```diff
@@ -23,23 +24,19 @@ const separator = filepath.Separator
 type Memory struct {
 	s *storage
 
-	m mutex
+	m sync.RWMutex
 }
 
 // New returns a new Memory filesystem.
 func New(opts ...Option) billy.Filesystem {
-	o := &options{
-		// Enable thread-safety by default.
-		newMutex: newMutex,
-	}
-
```

Comment:
Do we need `opts ...Option` and `options`?

Reply:
Yes, the plan is to align with the other osfs implementations in a future PR.